src/lib/structures/fpa_rounding.ml

```diff
@@ -88,7 +88,8 @@ let hstring_ae_reprs =
 (* The rounding mode is the enum with the SMT values.
    The Alt-Ergo values are injected in this type. *)
 let fpa_rounding_mode =
-  Ty.Tsum (Hs.make "RoundingMode", hstring_smt_reprs)
+  let fake_reprs = List.map Uid.of_hstring hstring_smt_reprs in
+  Ty.Tsum (Uid.of_string "RoundingMode", fake_reprs)
```

Comment:
Why use hstring rather than dolmen IDs here?

Reply:
When I wrote this PR  I didn't do it because I didn't want to have to manage unique identifiers in the legacy typechecker. Basically, if we use real unique identifiers and hstring identifiers, we have to ensure that the legacy typechecker won't recreate ADTs' constructors for builtin enums (and it does...). 

Finally I found another way to use `dolmen` identifiers here. I modified the comparison functions of `Uid`. Now if we compare a `Hstring` and a `Dolmen` identifiers, we transform the `Dolmen` identifier into string.

Reply:
> the legacy typechecker won't recreate ADTs' constructors for builtin enums (and it does...).

Ah, I guess this is the issue of having different names for the AE and SMT enums, got it.

> Now if we compare a Hstring and a Dolmen identifiers, we transform the Dolmen identifier into string.

Is this not re-introducing the dependency on string values that this PR is trying to eliminate?

Reply:
Now the `Dolmen` frontend doesn't produce `Uid.t` of type `Hstring` anymore (but for abstract types, I'll change that). It means we always compare `Dolmen` identifiers here (for constructors, destructors and ADTs' names).